File: POO/java/exercicios/Number.java
package exercicios;

public class Number {
	private int limit;
	private int value;
	
	public Number(int limit) {
		this.limit = limit;
		this.value = 0;
	}
	
	public int getValue() {
		return this.value;
	}
	
	public void setValue(int newValue) {
		this.value = newValue;
	}
	
	public String getDisplayValue() {
		if (this.value < 10) {
			return "0" + this.value;
		}
		else {
			return String.valueOf(this.value);
		}
	}
	
	public void increment() {
		this.value = (this.value + 1) % this.limit;
	}
}

File: POO/python/number.py
class Number:
    def __init__(self, limit):
        self.limit = limit
        self.value = 0

    def get_value(self):
        return self.value

    def set_value(self, new_value):
        self.value = new_value

    def get_display_value(self):
        if self.value < 10:
            return f"0{self.value}"
        else:
            return str(self.value)

    def increment(self):
        self.value = (self.value + 1) % self.limit
